src/action/common/place_nix_configuration.rs

```diff
@@ -28,65 +28,197 @@ impl PlaceNixConfiguration {
         extra_conf: Vec<String>,
         force: bool,
     ) -> Result<StatefulAction<Self>, ActionError> {
-        let extra_conf = extra_conf.join("\n");
-        let mut nix_config = nix_config_parser::NixConfig::parse_string(extra_conf, None)
-            .map_err(CreateOrMergeNixConfigError::ParseNixConfig)
+        let create_directory = CreateDirectory::plan(NIX_CONF_FOLDER, None, None, 0o0755, force)
+            .await
             .map_err(Self::error)?;
-        let settings = nix_config.settings_mut();
-
-        settings.insert("build-users-group".to_string(), nix_build_group_name);
-        let experimental_features = ["nix-command", "flakes", "auto-allocate-uids"];
-        match settings.entry("experimental-features".to_string()) {
-            Entry::Occupied(mut slot) => {
-                let slot_mut = slot.get_mut();
-                for experimental_feature in experimental_features {
-                    if !slot_mut.contains(experimental_feature) {
-                        *slot_mut += " ";
-                        *slot_mut += experimental_feature;
-                    }
-                }
-            },
-            Entry::Vacant(slot) => {
-                let _ = slot.insert(experimental_features.join(" "));
-            },
-        };
 
-        // https://github.com/DeterminateSystems/nix-installer/issues/449#issuecomment-1551782281
-        #[cfg(not(target_os = "macos"))]
-        settings.insert("auto-optimise-store".to_string(), "true".to_string());
+        let mut nix_conf_insert_settings = Vec::default();
+        nix_conf_insert_settings.push("include ./nix-installer-defaults.conf".into());
+        nix_conf_insert_settings.extend(extra_conf);
+        let nix_conf_insert_fragment = nix_conf_insert_settings.join("\n");
 
-        settings.insert(
-            "bash-prompt-prefix".to_string(),
-            "(nix:$name)\\040".to_string(),
-        );
+        let mut defaults_conf_settings = vec![
+            ("build-users-group", nix_build_group_name),
+            (
+                "experimental-features",
+                "nix-command flakes auto-allocate-uids".into(),
+            ),
+        ];
+        // https://github.com/DeterminateSystems/nix-installer/issues/449#issuecomment-1551782281
```

Comment:
I think this comment should go above the `auto-optimise-store` setting (down on line 60) rather than here.